devel/cmake-devel/Portfile

```diff
@@ -146,8 +146,13 @@ platform darwin {
 
 }
 
-# https://trac.macports.org/ticket/67943
 if {[string match *gcc* ${configure.compiler}]} {
+    # Utilities/cmlibuv/src/unix/fs.c:562:57: error: passing argument 4 of 'scandir'
+    # from incompatible pointer type [-Wincompatible-pointer-types]
+    configure.cflags-append \
+        -Wno-error=incompatible-pointer-types
```

Comment:
is this fixed/reported upstream already as this seems to be an real issue in the code, not something we just want to ignore. If so, please use the upstream patch/fix instead of this; if not, please report it.

Reply:
I strongly agree that real fixes are better than just shooting the messenger by removing the test.

Reply:
I believe, nobody here is aware of the context, so:
The bug is in `libuv` code. `libuv` upstream explicitly dumped anything earlier than a few latest MacOS. They won’t fix anything.
I will open an issue with CMake upstream today to see if they may be kind enough to fix `libuv` bug on their end and carry a patch.

Reply:
This has nothing to do with MacOS though. It's a complaint about incompatible pointer types.

Reply:
@pmetzger It has, because `scandir` is picked from macOS headers (or perhaps from LegacySupport in this case).

Reply:
Issue with CMake upstream: https://gitlab.kitware.com/cmake/cmake/-/issues/26223

Reply:
Don’t forget the scandir definition changed in 10.8 … this seems likely another example of that issue.

We have patched around that a number of times in the ports tree, although it would be nice if legacysupport could handle it.

Reply:
Indeed, if we deal with it in legacysupport, then it fixes many ports at once.